mlflow/cli.py

```diff
@@ -144,6 +144,20 @@ def run(uri, entry_point, version, param_list, experiment_name, experiment_id, m
         _logger.error("=== %s ===", e)
         sys.exit(1)
 
+def _validate_server_args(static_prefix=None, gunicorn_opts=None, workers=None, waitress_opts=None):
+    server_kwargs = {'static_prefix': static_prefix}
+    if sys.platform == "win32":
+        if gunicorn_opts is not None:
+            raise NotImplementedError("waitress replaces gunicorn on Windows, cannot specify --gunicorn-opts")
+        server_kwargs.update({
+            'waitress_opts': waitress_opts
+        })
+    else:
+        server_kwargs.update({
+            'workers': workers if workers else 1
+            'gunicorn_opts': gunicorn_opts
+        })
+    return server_kwargs
```

Comment:
It looks like the only purpose of this method is to check that gunicorn opts are not passed on windows (and maybe we want to check that waitress is not use on non-windows, at least for now). I think it would be simpler if we do not return anything in that case.